jbmc/unit/java_bytecode/java_bytecode_parser/parse_java_attributes.cpp

```diff
@@ -596,4 +597,24 @@ SCENARIO(
       }
     }
   }
+
+  GIVEN("A method that may or may not throw exceptions")
+  {
+    const symbol_tablet &new_symbol_table = load_java_class(
+      "ThrowsExceptions", "./java_bytecode/java_bytecode_parser");
+    WHEN("Parsing the exceptions attribute for a method that throws exceptions")
+    {
+      THEN("We should be able to get the list of exceptions it throws")
+      {
+        const symbolt &class_symbol =
+          new_symbol_table.lookup_ref("java::ThrowsExceptions");
+        const symbolt &method_symbol =
+          new_symbol_table.lookup_ref("java::ThrowsExceptions.test:()V");
+        const code_typet method = to_code_type(method_symbol.type);
+        const std::vector<irept> exceptions = method.throws_exceptions();
+        REQUIRE(id2string(exceptions.at(0).id()) == "CustomException");
+        REQUIRE(id2string(exceptions.at(1).id()) == "java.io.IOException");
```

Comment:
Can we actually rely on the ordering her and is the ordering actually important? Maybe checking if `exceptions` contains the exception which we are checking for, would be more robust.